### ISSUE TITLE
Batch: getting *all* results

### ISSUE THREAD

**Issue by ebrehault** [plone/plone.restapi#303]
b_size=-1 should return all results

**Comment by tisto** [comment]
@ebrehault What about b_size=0? Is using -1 a common pattern for this?

**Comment by ebrehault** [comment]
I think both exist. I am fine with 0 too.

**Comment by lukasgraf** [comment]
@ebrehault @tisto just as an FYI, the batching parameters like `b_size` and `b_start` are not only used by `plone.restapi`'s (hypermedia) batching implementation, but also passed on to the ZCatalog, which uses them for all kinds of stuff, for example to decide to do a more efficient [top-N sort](http://old.zope.org/Members/Caseman/ZCatalog_for_2.6.1/view) for small resultsets.

Some parts of the ZCatalog code (somewhat surprisingly if you're not used to it) also inspects the request directly in some places, so it will get its hands on parameters like `b_size` without us even passing them explicitly. That was part of the reason why I made sure to actually use the exact same parameter names in `plone.restapi` batching as in the rest of Plone - quite a few optimizations happen behind the scenes in respect to those parameters.

However, that also means a value like `-1` or `0` needs to be supported by that underlying codebase, not just `plone.restapi`.

Having said that, I agree with providing *some* way of getting all results of the batch. If they work, `-1` or `0` would both be fine with me.

So basically I'm just planting a sign post to think of the underlying codebase, for whomever is going to implement this (possibly future me) :wink:
